DOC-1248 Add slack input (#229)
redpanda-data/rp-connect-docs: modules/components/pages/inputs/slack.adoc

= slack
// tag::single-source[]
:type: input
:page-beta: true

component_type_dropdown::[]

> Connects to Slack using https://api.slack.com/apis/socket-mode[Socket Mode^], and can receive events, automated or user interactions, and slash commands.

Author: What is the use case for this connector. Maybe analytics or to automate certain workflows?

Author: Updated.

ifndef::env-cloud[]
Introduced in version 4.52.0.
endif::[]

```yml
# Common configuration fields, showing default values
input: 
  label: ""
  slack:
    app_token: "" # No default (required)
    bot_token: "" # No default (required)
    auto_replay_nacks: true
```
== Metadata

Each message emitted from this input has an `@type` metadata flag to indicate the event type, either `"events_api"`, `"interactions"`, or `"slash_commands"`.

== Fields

=== `app_token`

The app-level token to use to authenticate and connect to Slack.

*Type*: `string`

=== `bot_token`

> Your Slack bot user's OAuth token, which must have the correct permissions to access your Slack app's https://api.slack.com/methods/apps.connections.open[Socket Mode WebSocket URL^].

Author: This is probably wrong but what permissions/scopes does the OAuth token need?

Author: Updated.

= slack
// tag::single-source[]
:type: input

component_type_dropdown::[]

> Connects to Slack using https://api.slack.com/apis/socket-mode[Socket Mode^], and can receive events, interactions (automated and user initiated), and slash commands.

Reviewer: ```suggestion
Connects to Slack using https://api.slack.com/apis/socket-mode[Socket Mode^], and can receive events, interactions (automated and user-initiated), and slash commands.
```

Author: Updated.

This input is useful for:

- Building bots that can query or write data.
- Sending events to data warehouses.

> You could also try pairing this input with Redpanda Connect's AI processors. Search for processors with the prefix `cohere`, `openai`, and `ollama`.

Reviewer: ```suggestion
You could also try pairing this input with Redpanda Connect's AI processors, which use the prefix `cohere`, `openai`, and `ollama`.
```

Reviewer: Not sure if this is the exhaustive list of prefixes for AI processors? Your call of course.

Author: updated
